tox.ini

```diff
@@ -294,23 +292,41 @@ deps =
     {[testenv:build]deps}
     twine
 passenv =
+    # Intentionally omit TWINE_REPOSITORY_URL from the passenv list,
+    # as this overrides other ways of setting the repository and could
+    # unexpectedly result in releasing to the wrong repo
     {[testenv]passenv} \
     TWINE_USERNAME \
-    TWINE_PASSWORD \
-    TWINE_REPOSITORY_URL
+    TWINE_PASSWORD
 commands =
     {[testenv:build]commands}
-    twine upload --skip-existing --repository testpypi dist/*
```

Comment:
I would argue that the correct fix here is to remove `--repository testpypi`. That was a miss on my part when I initially populated out these changes.

The `--repository` flag identifies the profile in `~/.pypirc` that should be used. If you are specifying your username and repo url through environment variables, that flag is unnecessary.

Historical note: This and other Crypto Tools Python repos used be configured to use the `~/.pypirc` file for `twine` configuration. More recently I made changes to set the appropriate values via environment variables instead, both to remove unclear dependency on configuration that exists outside of the repo and to allow for future automation (ex passing a token in via GHS or `secrets-helper`).